.github/workflows/deploy-tests.yml

```diff
@@ -12,7 +12,7 @@ concurrency:  # Cancel previous workflows on the same pull request
 jobs:
 
   assess-file-changes:
-    uses: catalystneuro/neuroconv/.github/workflows/assess-file-changes.yml@main
+    uses: catalystneuro/neuroconv/.github/workflows/assess-file-changes.yml@add_support_to_python_312
```

Comment:
It's not necessary for the `assess-file-changes` workflow since it's not changed from `main`

The others steps modify the workflows from their versions on `main` (they add the "3.12" to the matrix), so the version that gets deployed needs to be the workflow version from this branch

We would change these back to say `main` prior to or post merge after we get proof that they pass successfully on 3.12

Reply:
Are you sure? Wouldn't the workflows deployed for a given PR be corresponding to the yaml files in the branch of that PR?

Reply:
I have the vague memory that I found similar problems in Spikeinterface and started using this practice (always be specific about the branch). 

| Are you sure? Wouldn't the workflows deployed for a given PR be corresponding to the yaml files in the branch of that PR?

That said, after I read @bendichter's question I tried to look for this in the CI documentation and I could not find where the behavior of @ref in `uses` is documented without luck.  

I have the hunch that to get @bendichter expected behavior we should use relative paths for re-using workflows:

https://docs.github.com/en/actions/using-workflows/reusing-workflows#calling-a-reusable-workflow

It seems that the default behavior when you use absolute paths the way that we use know is that it would pull for the main branch.

Reply:
> Are you sure?

100% positive when using absolute paths; it points directly to the file that would exist at the location if you prepended 'https://github.com/' to the absolute path, and the @ controlling the branch

Try the relative paths now in https://github.com/catalystneuro/neuroconv/pull/626/commits/6654a0eda980199763a0ecd5111fe63ce5fd42ce, hopefully that works and would fix this annoyance - thanks @h-mayorquin for the pointer

Reply:
Relative seems to be working well!

Reply:
Great, that would make PR related to CI changes easier : D

Reply:
ah, gotcha. Glad you could find a solution that works for relative path. This is a nice simplification of the process